saleor/core/db/expressions.py

```diff
@@ -0,0 +1,49 @@
+from django.db.models import JSONField
+from django.db.models.expressions import Expression
+
+
+class JsonCat(Expression):
+    # It's possible to recursevily create that expression to allow to pass
```

Comment:
One more not on top of that, if we find that we want more operators to be implemented this way, we can make generic class for that solution and use inheritance mechanism later on.

I wanted to stick to the simple solution for this proof of concept, including only dealing with two arguments.